python-package/lightgbm/basic.py

```diff
@@ -1297,6 +1475,74 @@ def _lazy_init(self, data, label=None, reference=None,
         # set feature names
         return self.set_feature_name(feature_name)
 
+    def __yield_row_from(self, seqs, indices):
+        # type: (List[Sequence], Iterable[int]) -> ...
+        offset = 0
+        seq_id = 0
+        seq = seqs[seq_id]
+        for row_id in indices:
+            assert row_id >= offset, "sample indices are expected to be monotonic"
+            while row_id >= offset + len(seq):
+                offset += len(seq)
+                seq_id += 1
+                seq = seqs[seq_id]
+            id_in_seq = row_id - offset
+            row = seq[int(id_in_seq)]
+            yield row
```

Comment:
TODO: fix unwanted memory view reference

Reply:
Either user or LGBM should be doing for once.
Are LGBM users generally trusted in design philosophy?